### PR TITLE
ROC-6905 ROC-7726 Add feature toggle

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link (if applicable) ###
Parent: https://tools.hmcts.net/jira/browse/ROC-6905
This subtask: https://tools.hmcts.net/jira/browse/ROC-7726

### Change description ###
Add configuration option `feature_toggles.auto_cancel_payments`
Import via env var `FEATURE_AUTO_CANCEL_PAYMENTS`
Set to `false` on production, `true` elsewhere

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java [hunk 1]
```diff
@@ -110,8 +113,12 @@ public Payment createPayment(
     }
 
     public void cancelPayment(String authorisation, String paymentReference) {
-        logger.info("Cancelling payment {}", paymentReference);
-        paymentsClient.cancelPayment(authorisation, paymentReference);
+        if (autoCancel) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java]: My preference would be if this flag is off don't even call the method `cancelPayment`.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java [hunk 2]
```diff
@@ -43,7 +44,8 @@ public PaymentsService(
         @Value("${payments.api.service}") String service,
         @Value("${payments.api.siteId}") String siteId,
         @Value("${payments.api.currency}") String currency,
-        @Value("${payments.api.description}") String description
+        @Value("${payments.api.description}") String description,
+        @Value("${feature_toggles.auto_cancel_payments}") boolean autoCancel
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java]: Probably not for this PR but having a compose object for all the configuration properties of feature_toggles would be more cleaner. And we can inject the object `FeatureToggles` i.e

```suggestion
        FeatureToggles  featureToggles
```

```suggestion

@Validated
@Component
@ConfigurationProperties(prefix = "feature_toggles")
public class FeatureToggles { 
     @NotEmpty
     private String autoCancelPayments;
     ...//other feature toggles here 
        }
```

Comment (andrewwa-kainos) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java]: Nice idea, wanna raise a tech-debt for that @sabahirfan ?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java (hunk 1 above)
Comment (andrewwa-kainos) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java]: Agreed. See https://github.com/hmcts/cmc-claim-store/pull/1587/commits/a6055e9948254eccdabd140bed6e40e9d8892320

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java (hunk 2 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java]: yes please. 👍

File: infrastructure/variables.tf [hunk 3]
```diff
@@ -107,3 +107,7 @@ variable schedule_state-transition_waiting-transfer  {
 variable "enable_ase" {
   default = false
 }
+
+variable "auto_cancel_payments" {
+  default = true
```
Comment (ckalista) [inline infrastructure/variables.tf]: Should we not be conservative and set the default to false for now?

**Pull request merged**